TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,2695 @@
+            PropagateProperties(nestedCfg);
+            this.CurrentProgramCfgBuilder.AllCfgBuilder.Add(nestedCfg);
+            this.CurrentProgramCfgBuilder = nestedCfg;
+            this.CurrentProgramCfgBuilder.InitializeCfg(funDecl);
+            this.CurrentProgramCfgBuilder.CurrentProgramCfgBuilder = this.CurrentProgramCfgBuilder;
+        }
+
+        /// <summary>
+        /// Leave a function declaration
+        /// </summary>
+        /// <param name="funDecl">Function declaration left</param>
+        protected virtual void LeaveFunction(FunctionDeclaration funDecl)
+        {
+            this.CurrentProgramCfgBuilder = this.CurrentProgramCfgBuilder.ParentProgramCfgBuilder;
+        }
+
+
+        /// <summary>
+        /// Enter in the domain a section or a paragraph symbol.
+        /// </summary>
+        /// <param name="sym">The symbol to enter.</param>
+        private void EnterSectionOrParagraphSymbol(Symbol sym)
+        {
+            System.Diagnostics.Debug.Assert(sym.Kind == Symbol.Kinds.Section || sym.Kind == Symbol.Kinds.Paragraph);
+
+            if (this.CurrentProgramCfgBuilder.SectionsParagraphs == null)
+                this.CurrentProgramCfgBuilder.SectionsParagraphs = new Container<Symbol>();
+            this.CurrentProgramCfgBuilder.SectionsParagraphs.Add(sym);
+
+            if (this.CurrentProgramCfgBuilder.AllSectionsParagraphs == null)
+                this.CurrentProgramCfgBuilder.AllSectionsParagraphs = new List<Symbol>();
+            sym.Number = this.CurrentProgramCfgBuilder.AllSectionsParagraphs.Count;
+            this.CurrentProgramCfgBuilder.AllSectionsParagraphs.Add(sym);
+
+            //Special case Section or Paragraph inside a Declarative
+            if (this.CurrentProgramCfgBuilder.CurrentDeclarativesContext != null)
+            {
+                switch (sym.Kind)
+                {
+                    case Symbol.Kinds.Paragraph:
+                        {
+                            CfgParagraphSymbol cfgPara = (CfgParagraphSymbol)sym;
+                            cfgPara.SetFlag(Symbol.Flags.Declaratives, true);
+                        }
+                        break;
+                    case Symbol.Kinds.Section:
+                        {
+                            CfgSectionSymbol cfgSymbol = (CfgSectionSymbol)sym;
+                            cfgSymbol.SetFlag(Symbol.Flags.Declaratives, true);
+                            this.CurrentProgramCfgBuilder.CurrentDeclarativesContext.AddSection(cfgSymbol);
+                        }
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Resolve a section or a paragraph symbol reference
+        /// </summary>
+        /// <param name="symRef">The Symbol Reference instance to a section or a paragraph.</param>
+        /// <returns>The scope of symbols found</returns>
+        internal Container<Symbol>.Entry ResolveSectionOrParagraphSymbol(SymbolReference symRef)
```

Comment:
This a new Issue, because current CrossCheck does not support what CFG has implemented. This is not a refactorization, but a port to CrossChecker to use either what CFG does or what SDLite will implement to support also qualified Section and Paragraph symbol resolution. So that CFG can move to the port or SDLite implementation, because CFG has every thing to do that.

Reply:
Done thanks to #1736, CFG and CrossCheck now use the same method which also supports qualified name resolution.